src/uthash.h

```diff
@@ -105,6 +102,30 @@ typedef unsigned char uint8_t;
 #define uthash_expand_fyi(tbl)            /* can be defined to log expands   */
 #endif
 
+#ifndef HASH_NOMEM_OK
+#define HASH_NOMEM_OK 0
+#endif
+
+#if HASH_NOMEM_OK
+/* malloc failures can be recovered from */
+
+#ifndef uthash_fatal
+#define uthash_fatal(obj) do{}while(0)
```

Comment:
It looks like you're overloading the name `uthash_fatal` to mean two very different things:

- If `!HASH_NOMEM_OK`, it should terminate the program, takes a string literal, and is called on every OOM.
- If `HASH_NOMEM_OK`, it need not terminate the program, takes a hash element, and is called only on OOMs that happen inside `HASH_SELECT`.

In the second case, `uthash_fatal` is the wrong name because it's *not* called under unrecoverable circumstances and it's *not* required to be "fatal" to the program. But `uthash_oom` would also be a wrong name, so long as we're not calling it on every OOM. Now, I think you *should* call it on every OOM — that would suitably fix that problem, and then we could plausibly name it `uthash_oom(obj, where)`.

I think `uthash_oom` should take `(obj, where)` for the same reason that `HASH_FSCK` takes `where`: it might help the programmer to debug the issue.